pkg/ovs/openflow/testing/utils.go

```diff
@@ -0,0 +1,1167 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testing
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"antrea.io/libOpenflow/openflow15"
+	"antrea.io/libOpenflow/util"
+	"antrea.io/ofnet/ofctrl"
+	"k8s.io/klog/v2"
+)
+
+var TableNameCache map[uint8]string
+
+type fieldMetadata struct {
+	name   string
+	length uint8
+}
+
+func (m *fieldMetadata) getMatchNickname() string {
+	name := strings.TrimPrefix(m.name, "NXM_OF_")
+	name = strings.TrimPrefix(name, "NXM_NX_")
+	name = strings.TrimPrefix(name, "OXM_OF_")
+	name = strings.ToLower(name)
+	if strings.HasPrefix(name, "ip_") && name != "ip_dscp" {
+		name = strings.Replace(name, "ip_", "nw_", 1)
+	}
+	if strings.HasPrefix(name, "ipv4_") && name != "ip_dscp" {
+		name = strings.Replace(name, "ipv4_", "nw_", 1)
+	}
+	if strings.HasPrefix(name, "icmpv6_") {
+		name = strings.Replace(name, "icmpv6_", "icmp_", 1)
+	}
+	if strings.HasPrefix(name, "tun_ipv4_") {
+		name = strings.Replace(name, "tun_ipv4_", "tun_", 1)
+	}
+	if strings.HasPrefix(name, "eth_") {
+		name = strings.Replace(name, "eth_", "dl_", 1)
+	}
+	if strings.HasPrefix(name, "tcp_") {
+		name = strings.Replace(name, "tcp_", "tp_", 1)
+	}
+	if strings.HasPrefix(name, "udp_") {
+		name = strings.Replace(name, "udp_", "tp_", 1)
```

Comment:
This only works for TCP src/dst port, but does not work for tcp_flags. Maybe add comments here to avoid unexpected failure?

Reply:
I'll change them to 
```
if name == "tcp_dst" || name == "tcp_src" {
		name = strings.Replace(name, "tcp_", "tp_", 1)
	}
	if name == "udp_dst" || name == "udp_src"{
		name = strings.Replace(name, "udp_", "tp_", 1)
	}
```